sentry-core/src/main/java/io/sentry/core/UncaughtExceptionHandlerIntegration.java

```diff
@@ -83,4 +84,11 @@ static Throwable getUnhandledThrowable(Thread thread, Throwable thrown) {
     mechanism.setType("UncaughtExceptionHandler");
     return new ExceptionMechanismThrowable(mechanism, thrown, thread);
   }
+
+  @Override
+  public void close() {
+    if (defaultExceptionHandler != null) {
+      threadAdapter.setDefaultUncaughtExceptionHandler(defaultExceptionHandler);
```

Comment:
This is problematic. We need to check if we are still the current handler before we go ahead and override the current one.
After we set ourselves there, another integration might have set themselves and wrapped us.